### ISSUE TITLE
Documentation: clarify test example in part 3 of starting guide

### ISSUE THREAD

**Issue by FlorianGit** [akka/akka#25086]
I'm currently working through [part 3](https://doc.akka.io/docs/akka/current/guide/tutorial_3.html#testing-the-actor) of the starting guide, "testing the actor" and had a bit of trouble getting the code to work. The example leaves out the details of exactly what kind of spec is used.

I had a look at the [source of the snippet](https://github.com/akka/akka/blob/master/akka-docs/src/test/scala/tutorial_3/DeviceSpec.scala) but could not get it to work either because the AkkaSpec used there [is not part of the released version](https://stackoverflow.com/a/33629136/5367428) of the akka Testkit.

Fortunately the source of the akka testkit is also available and I managed to build a working test using

    class DeviceSpec extends TestKit(ActorSystem("DeviceSpec")) 
        with Matchers 
        with WordSpecLike 
        with BeforeAndAfterAll {

I think the readability of this part of the documentation can be improved by adding the complete required source for the first test ("reply with empty reading if no temperature is known"). For the second test, "reply with latest temperature reading" the snippet can stay as it is, with the remark that it should be added to the previous spec.

So my question is what you think of this idea? I do not mind creating a pull request implementing it, but first want to check if it would be desirable.

Thanks,
Florian

**Comment by johanandren** [comment]
The reason why we don't pull in the entire sample is to keep the sample short and somewhat test framework agnostic (although we are perhaps already using Scalatest-only test syntax here).

I don't think we want to pull in the complete sample here, as it it would make a simple "here's a minimal test case" take so much space/attention. The sample project referenced in the beginning of the getting started tutorial should also already contain a more complete sample that does provide everything you need? (https://doc.akka.io/docs/akka/current/guide/tutorial.html#prerequisites)

**Comment by johanandren** [comment]
Or include the complete test in the sample project? (if it isn't already)

**Comment by FlorianGit** [comment]
Yes, you're right, I started from scratch instead of from the sample project (although I did have the sample project locally as a reference). The guide does mention in the introduction that it's best to add the functionalities introduced in the guide to the sample project, so that part was my own doing. 

However, I do think it can be an improvement if the reader of the documentation can resort to just copy pasting to make things work. The test in the sample project uses a `FlatSpecLike`, whereas the guide uses test formulations in the style of `AkkaSpec`, which uses `WordSpecLike`. 

So in name of consistency (while preserving the flow of the document), I think either of the following could be nice:
 1. Change the sample project to use `WordSpecLike`, so that the examples from the guide can be easily pasted into the sample project
 2.  Replace the definition of the `DeviceSpec` in the [snippet](https://github.com/akka/akka/blob/master/akka-docs/src/test/scala/tutorial_3/DeviceSpec.scala) to extend a `Testkit with Matchers with WordSpecLike with BeforeAndAfterAll {
`

I must add that in total I found the guide great and well written, so thanks for providing that. I don't mind creating a pull request for either of the above, but I can also imagine that you think it's not worth the trouble, since it is a small issue anyway, so you can also close the issue if you prefer.

**Comment by raboof** [comment]
@FlorianGit yes, thanks for clarifying how you got there - even though we recommend starting with the quickstart project I think there will be many users who, like you, prefer to start from scratch, and we should make that as easy as possible as well.

I think I'd agree with either change (perhaps even both). Do we remember why we went with `FlatSpecLike` for the quickstart project (perhaps @henrikengstrom?)

**Comment by FlorianGit** [comment]
I created a [pull request](https://github.com/akka/akka-quickstart-scala.g8/pull/19) for the first point. 

Regarding the second point I realised that the `AkkaSpec` is used in ~60 places in the documentation files. Changing it only in the place I encountered feels arbitrary. However, changing it in every place might not be correct, since the AkkaSpec defines more than only the `TestKit` extended with the traits I mentioned earlier. Since all files there are snippets and as far as I know not runnable, I don't see an easy way to verify that everything would  be correct if I were to perform a massive search/replace.
